tests/functional/secret/test_scan_repo.py

```diff
@@ -101,3 +101,54 @@ def test_scan_repo_exclude_patterns(
     )
 
     assert "No secrets have been found" in result.stdout
+
+
+@pytest.mark.parametrize("ignore_removed_secrets", [True, False])
+@pytest.mark.parametrize("file_deleted", [True, False])
+def test_scan_repo_ignore_removed_secrets(
+    tmp_path_factory: pytest.TempPathFactory,
```

Comment:
The `leaky_repo()` fixture uses `tmp_path_factory` because it's a module-scope fixture, but since this test creates its own repo, it can use `tmp_path`.